### DISCUSSION TITLE
Chosen option from relationship picker cannot be retrieved

### DISCUSSION THREAD

**Discussion by starmomou** [Budibase/budibase#3306] [Help] [answered]
Hi everyone,

There are two main tables in my app: **Staff Info** and **On Behalf Of**. The situation is quite similar to #2548 and I created a many-to-one relationship between them.

<img width="1071" alt="螢幕截圖 2021-11-10 上午10 29 41" src="https://user-images.githubusercontent.com/73826858/141041126-b43f33bc-1d93-4c7c-b9db-ca81af4716f8.png">

I can choose an option (11, 22, 33) from **On Behalf Of** table using a _relationship picker_. 
<img width="1202" alt="螢幕截圖 2021-11-10 上午11 16 53" src="https://user-images.githubusercontent.com/73826858/141044265-78fd21d6-c0ea-440b-b629-55f073c9219e.png">

However, whenever I click the submit button. Everything can be saved correctly except the **On Behalf Of** column unless I directly create row into **Staff Info** table.
<img width="1398" alt="螢幕截圖 2021-11-10 上午10 30 15" src="https://user-images.githubusercontent.com/73826858/141041554-bf84632c-5653-41cb-99ff-62369bef1076.png">

Is there something wrong in my Save Row action? I only have one _Validate Form_ using **Staff Info** schema for all fillable blanks.
<img width="1039" alt="螢幕截圖 2021-11-10 上午11 08 09" src="https://user-images.githubusercontent.com/73826858/141042730-cffe4d1e-ee32-42b5-9878-6f2e0d9c984f.png">

One more little thing, can I rename the action **Save Row** to anything else? This is too long to read...😢 I can provide more details if something is unclear. Thank you.
<img width="977" alt="螢幕截圖 2021-11-10 上午11 27 33" src="https://user-images.githubusercontent.com/73826858/141044701-2826d83a-1937-4220-9757-0bfd566524f9.png">

**Comment by mjashanks** [comment, marked as answer]
For your `On Behalf Of` relationship - I don't think you should need to add it as a column in the save row action. It is already inside your form - so I would try to simply delete this:

![image](https://user-images.githubusercontent.com/3524181/141085935-d3fff6d3-a933-482e-8c0e-6f03f9686a8a.png)

FYI - the "Add Column" is there so that you can set the column value to something that is **not** entered within the form. For example, you might want to calculate a value in javascript.

On the other hand - I would have expected that to work anyway.

For the `Save Row` rename - unfortunately not. I definitely see the need for this though.

**Reply by starmomou** [reply]
Thank you so much for the help!
